From 70916cf6a67c8c2557c8cd6a4f5862398c94e7f2 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Mon, 13 Oct 2025 11:37:06 +0800
Subject: [PATCH 41/63] loonggpu: Get rid of ida_simple_get and
 ida_simple_remove

They'd been a deprecated wrapper of ida_alloc_range and ida_free since
Linux 5.10, and they were finally removed in 6.18-rc1.

Call ida_alloc_range and ida_free directly to fix build failure.

Link: https://git.kernel.org/torvalds/c/3264ceec8f17
Link: https://git.kernel.org/torvalds/c/c8a09fc9664f
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_ids.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/loonggpu/loonggpu_ids.c b/loonggpu/loonggpu_ids.c
index d53ee33..6356121 100644
--- a/loonggpu/loonggpu_ids.c
+++ b/loonggpu/loonggpu_ids.c
@@ -40,9 +40,9 @@ int loonggpu_pasid_alloc(unsigned int bits)
 	int pasid = -EINVAL;
 
 	for (bits = min(bits, 31U); bits > 0; bits--) {
-		pasid = ida_simple_get(&loonggpu_pasid_ida,
-				       1U << (bits - 1), 1U << bits,
-				       GFP_KERNEL);
+		pasid = ida_alloc_range(&loonggpu_pasid_ida,
+					1U << (bits - 1), (1U << bits) - 1,
+					GFP_KERNEL);
 		if (pasid != -ENOSPC)
 			break;
 	}
@@ -60,7 +60,7 @@ int loonggpu_pasid_alloc(unsigned int bits)
 void loonggpu_pasid_free(unsigned int pasid)
 {
 	trace_loonggpu_pasid_freed(pasid);
-	ida_simple_remove(&loonggpu_pasid_ida, pasid);
+	ida_free(&loonggpu_pasid_ida, pasid);
 }
 
 static void loonggpu_pasid_free_cb(struct dma_fence *fence,
-- 
2.52.0

